src/Razor/src/Microsoft.AspNetCore.Razor.OmniSharpPlugin/DefaultProjectChangePublisher.cs

```diff
@@ -91,15 +91,15 @@ protected virtual void SerializeToFile(OmniSharpProjectSnapshot projectSnapshot,
             using (var writer = tempFileInfo.CreateText())
             {
                 _serializer.Serialize(writer, projectSnapshot);
+            }
 
-                var fileInfo = new FileInfo(publishFilePath);
-                if (fileInfo.Exists)
-                {
-                    fileInfo.Delete();
-                }
+            var fileInfo = new FileInfo(publishFilePath);
+            if (fileInfo.Exists)
+            {
+                fileInfo.Delete();
```

Comment:
Possible this throws due to being simultaneously read?

Reply:
I don't think so because the read and the delete take separate locks. I could be wrong though.